sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -240,6 +240,9 @@ statement
     | MSCK REPAIR TABLE multipartIdentifier                            #repairTable
     | op=(ADD | LIST) identifier (STRING | .*?)                        #manageResource
     | SET ROLE .*?                                                     #failNativeCommand
+    | SET TIME ZONE interval                                           #setTimeZone
+    | SET TIME ZONE timezone=(STRING | LOCAL)                          #setTimeZone
+    | SET TIME ZONE .*?                                                #setTimeZone
```

Comment:
It sounds like none of the systems are following ANSI SQL syntax. 

- https://www.ibm.com/support/knowledgecenter/SSEPEK_10.0.0/sqlref/src/tpc/db2z_sql_setsessiontimezone.html
- https://www.postgresql.org/docs/12/sql-set.html
- https://dev.mysql.com/doc/refman/8.0/en/time-zone-support.html
- https://docs.oracle.com/cd/E11882_01/server.112/e10729/ch4datetime.htm#NLSPG263

Reply:
We are close to the DB2 syntax, except that we support interval and `LOCAL`, and we don't allow the optional `SESSION` keyword.